src/connector/src/source/base.rs

```diff
@@ -397,13 +412,22 @@ impl ConnectorProperties {
 }
 
 impl ConnectorProperties {
-    pub fn extract(mut with_properties: HashMap<String, String>) -> Result<Self> {
+    /// Creates typed source properties from the raw `WITH` properties.
+    ///
+    /// It checks the `connector` field, and them dispatches to the corresponding type's `try_from_hashmap` method.
+    ///
+    /// `deny_unknown_fields`: Since `WITH` options are persisted in meta, we do not deny unknown fields when restoring from
+    /// existing data to avoid breaking backwards compatibility. We only deny unknown fields when creating new sources.
+    pub fn extract(
```

Comment:
I think it is better to add a new interface that will deny unknown fields. e.g. `extract_known_fields`

Reply:
Why it's better? I think with current interface, we are forced to think about the behavior whenever we want to create `ConnectorProperties`